src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -77,50 +82,134 @@ class CrossDeviceLink extends Component {
   }
 
   render = () =>
-    this.props.roomId ? <CrossDeviceLinkUI roomId={this.props.roomId} /> : <Spinner />
+    this.props.roomId ? <CrossDeviceLinkUI {...this.props}/> : <Spinner />
 }
 
 class CrossDeviceLinkUI extends Component {
   constructor(props) {
     super(props)
-    this.state = {copySuccess: false}
+    this.state = {
+      copySuccess: false,
+      sending: false,
+      mobileNumber: null,
+      error: {}
+    }
   }
 
+  linkCopiedTimeoutId = null
+
   copyToClipboard = (e) => {
+    e.preventDefault()
     this.textArea.select()
     document.execCommand('copy')
     e.target.focus()
     this.setState({copySuccess: true})
+    this.clearLinkCopiedTimeout()
+    this.linkCopiedTimeoutId = setTimeout(() => {
+      this.setState({copySuccess: false})
+    }, 5000)
+  }
+
+  clearLinkCopiedTimeout = () => {
+    if (this.linkCopiedTimeoutId) {
+      clearTimeout(this.linkCopiedTimeoutId)
+    }
+  }
+
+  updateNumber = (mobileNumber) => {
+    this.setState({mobileNumber})
+  }
+
+  clearPreviousAttempts = () => {
+    if (this.hasNumberError()) this.clearError()
+    if (this.state.mobileNumber) this.clearNumber()
+  }
+
+  clearNumber = () => {
+    this.setState({mobileNumber: null})
+  }
+
+  clearError = () => {
+    this.setState({error: {}})
+  }
+
+  setError = (name, type) => this.setState({error: {name, type}})
+
+  handleResponse = (response) => {
+    this.setState({sending: false})
+    if (response.status === "OK") {
+      this.props.actions.setMobileNumber(this.state.mobileNumber)
+      this.props.nextStep()
+    }
+    else {
+      this.setError('SMS_FAILED', 'error')
```

Comment:
Do we need a second argument here - it seems redundant given the function is called `setError`

Reply:
I am reusing the Error component an error can be an `error` or a `warn`. This one is an `error`. So this is consistent with the way we pass the errors in the other screens. Maybe we should rename the error types, but I don't think it's within the scope of this ticket.

Reply:
Yes, but the `setError` function should always set the type to error so you don't need to pass it as an argument

Reply:
I was also using to set errors without types, ie FE errors that don't have a type, but I can change that

Reply:
Can you remove the second argument here?